unblob/handlers/compression/gzip.py

```diff
@@ -124,3 +147,33 @@ def calculate_chunk(self, file: File, start_offset: int) -> Optional[ValidChunk]
             start_offset=start_offset,
             end_offset=file.tell(),
         )
+
+
+class MultiVolumeGzipHandler(DirectoryHandler):
+    NAME = "multi-gzip"
+    EXTRACTOR = MultiGZIPExtractor()
+
+    PATTERN = Glob("*.gz.*")
+
+    def calculate_multifile(self, file: Path) -> Optional[MultiFile]:
```

Comment:
With the generalization of `PATTERN` to cover all `split` output, `calculate_multifile()` would generate many potential `MultiFile` root files, including `file.gz.01`, `file.gz.ab`, ... resulting in potential conflicts between them (given some other conditions explained below).

Therefore this methods needs to be more picky, and a `MultiFile` should be recognized *only* if the input `file` `Path` is the first of the gzip parts - and has a gzip header.

For parts generated with `split`ting up a `gzip`ped file, it is usually not a problem, because we are filtering out all the other roots, and they are unlikely to have a valid gzip header.

However, https://www.gnu.org/software/gzip/manual/html_node/Advanced-usage.html mentions, that

> Multiple compressed files can be concatenated.

Which means, that parts can be generated with `split`ting the input first, then `gzip`ping the resulting parts (potentially in parallel for time-optimized builds), which would result in slightly different sized part files, but which all have valid gzip headers, these parts would all generate conflicting `MultiFile`-s with the current code.

----

We might also probably be prepared to exclude non-part files from `MultiFile.paths` like `file.gz.md5`, `file.gz.sha1`, `file.gz.txt`, `file.gz.README`, etc. here.

Reply:
Agree. I don't think we need to make a special case for files with `.md5` or `.README` extensions since the input file provided to `7z` is always the first one in the sorted list and `7z` has its own logic to process split content.

I'll create integration test files to cover the different scenarios you mention here and work from there.

Reply:
Done !